src/components/project/dto/project-transition.dto.ts

```diff
@@ -0,0 +1,46 @@
+import { Field, InputType, ObjectType } from '@nestjs/graphql';
+import { keys as keysOf } from 'ts-transformer-keys';
+import { DbLabel, ID, IdField, Secured, SecuredProps } from '../../../common';
+import { User } from '../../user/dto';
+import { IProject, Project } from './project.dto';
+import { ProjectStep } from './step.enum';
+
+@InputType()
+export abstract class ProjectTransitionInput {
```

Comment:
I was hoping to resolve the "transition" vs "step change" phrase before but it didn't happen.
I talked with @sethmcknight more and I think "transition" is too ambiguous.
Can we change all of those to StepChange? That makes it consistent with other half of the new types.